### PR TITLE
MLIBZ-1593 Don't open two databases to check support

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
In order to check support for a database adapter it is needed to open the database and save an item. This change opens a database with the same name as the database used to store data for the application. This will prevent opening two databases with a size of 5mb requiring a user to click accept on a popup to allow data to be stored.

#### Changes
- add a `loadAdapter()` function to all storage adapters
- Use the name of the application database to check for support of an adapter

### CODE REVIEW

File: src/middleware/src/storage/src/indexeddb.js [hunk 1]
```diff
@@ -273,28 +273,31 @@ export default class IndexedDB {
...
-      return Promise.resolve(isSupported);
+    if (typeof indexedDB === 'undefined') {
+      isSupported = false;
+      return Promise.resolve(undefined);
     }
 
-    const db = new IndexedDB(name);
-    return db.save(name, { _id: '1' })
-      .then(() => db.clear())
+    return db.save('__testSupport', { _id: '1' })
```
Comment (tejasranade) [inline src/middleware/src/storage/src/indexeddb.js:293]: Any particular reason to change this? It's better to keep it the same, don't you think? If I'm already using the SDK, will I see 2 DB entries after this change?

File: src/middleware/src/storage/src/websql.js [hunk 2]
```diff
@@ -7,20 +7,20 @@ import isFunction from 'lodash/isFunction';
...
 const idAttribute = process.env.KINVEY_ID_ATTRIBUTE || '_id';
 const masterCollectionName = 'sqlite_master';
-const size = 5 * 1000 * 1000; // Database size in bytes
 let dbCache = {};
-let isSupported = undefined;
+let isSupported;
 
 export default class WebSQL {
-  constructor(name = 'kinvey') {
+  constructor(name = 'kinvey', size = 5000) {
```
Comment (tejasranade) [inline src/middleware/src/storage/src/websql.js]: Why did size move to this method? I actually like that it's a constant. Easier to find in our shipped JS file. 
Also, why the change in default to 5000 from 5,000,000?

File: src/middleware/src/storage/src/websql.js [hunk 3]
```diff
@@ -7,20 +7,20 @@ import isFunction from 'lodash/isFunction';
...
     this.name = name;
+    this.size = size;
   }
 
   openDatabase() {
     let db = dbCache[this.name];
 
     if (!db) {
-      db = global.openDatabase(this.name, 1, '', size);
+      db = global.openDatabase(this.name, 1, '', this.size);
```
Comment (tejasranade) [inline src/middleware/src/storage/src/websql.js]: The third parameter here is used by the browser as the database name in the prompt to the user. I'm wondering what name makes sense there, but leaving it blank looks very odd (check the screenshot in MLIBZ-1593)

**Review by tejasranade — CHANGES_REQUESTED** [review]
Looks ok. Other than the notes below, can you comment on the tests performed on this change? How much of this change is covered by the test suite?

File: src/middleware/src/storage/src/websql.js (hunk 2 above)
Comment (thomasconner) [reply src/middleware/src/storage/src/websql.js]: I initially thought to allow to pass a different size if needed with a default of 5mb. I can movie it back to a `CONST` if that makes more sense.

File: src/middleware/src/storage/src/websql.js (hunk 3 above)
Comment (thomasconner) [reply src/middleware/src/storage/src/websql.js]: Good point. I am not sure what name to provide. "Kinvey" possibly?

File: src/middleware/src/storage/src/indexeddb.js (hunk 1 above)
Comment (thomasconner) [reply src/middleware/src/storage/src/indexeddb.js:293]: Yes it is possible you could see 2 DB entities. But in order to not show the popup we can't open 2 DB entries. This instead opens the DB used to store application data to also test support for the adapter.

**Comment by thomasconner** [comment]
All testing is manual at the moment. I need to work on the integration tests which would allow us to test this with code. We discussed setting aside time in the next sprint to work on integration tests.

File: src/middleware/src/storage/src/websql.js (hunk 2 above)
Comment (tejasranade) [reply src/middleware/src/storage/src/websql.js]: Having it parameterized is fine, but I didn't find the default easily. The way it was written previously (as a `const`) allowed me to quickly identify the default value.

File: src/middleware/src/storage/src/websql.js (hunk 3 above)
Comment (tejasranade) [reply src/middleware/src/storage/src/websql.js]: Since it gets prompted to the end user, I'm a little unsure (the end user presumably doesn't know what Kinvey is). Maybe something like "Local Cache" or "App Cache", which indicates the purpose of the database?

File: src/middleware/src/storage/src/indexeddb.js (hunk 1 above)
Comment (tejasranade) [reply src/middleware/src/storage/src/indexeddb.js:293]: Sorry, my initial question wasn't clear. I was asking about the change of name from `testIndexedDBSupport` to `__testSupport`. Whether that change will result in 2 entries for older Kinvey apps (one for each name).
I don't think it's a big deal, but it's worth noting that the change potentially affects older clients.

File: src/middleware/src/storage/src/websql.js [hunk 4]
```diff
@@ -7,9 +7,9 @@ import isFunction from 'lodash/isFunction';
 import isString from 'lodash/isString';
 const idAttribute = process.env.KINVEY_ID_ATTRIBUTE || '_id';
 const masterCollectionName = 'sqlite_master';
-const size = 5 * 1000 * 1000; // Database size in bytes
 let dbCache = {};
-let isSupported = undefined;
+let isSupported;
+const SIZE = 5000;
```
Comment (tejasranade) [inline src/middleware/src/storage/src/websql.js]: @thomasconner Again, why `5000`? It was `5 * 1000 * 1000` before. Am I missing something?

**Review by tejasranade — CHANGES_REQUESTED** [review]

File: src/middleware/src/storage/src/indexeddb.js (hunk 1 above)
Comment (thomasconner) [reply src/middleware/src/storage/src/indexeddb.js:293]: They could possibly have 2 entries. `testIndexedDBSupport` was used in a completely separate DB but it won't be an issue.

File: src/middleware/src/storage/src/websql.js (hunk 4 above)
Comment (thomasconner) [reply src/middleware/src/storage/src/websql.js]: I meant to put it back to what it was before.
